realm/src/main/java/io/realm/RealmQuery.java

```diff
@@ -172,32 +173,35 @@ private boolean containsDot(String s) {
     }
 
     /**
-     * Test if a field is null. Only works for relationships and RealmLists.
+     * Test if a field is {@code null}. Only works for not required types.
      *
-     * @param fieldName - the field name
-     * @return The query object
-     * @throws java.lang.IllegalArgumentException if field is not a RealmObject or RealmList
+     * @param fieldName - the field name.
+     * @return The query object.
+     * @throws java.lang.IllegalArgumentException if the field is annotated with @{@link io.realm.annotations.Required}
```

Comment:
If the field is not nullable. RealmLists, primitive types or boxed types with `@{@link io.realm.annotations.Required}` are not nullable.